### ISSUE TITLE
Test errors on Firefox

### ISSUE THREAD

**Issue by dmarcos** [aframevr/aframe#5326]
https://github.com/aframevr/aframe/actions/runs/5468426408/jobs/9956050718

weird one. That happens when adding a new commit but re-running the tests sometimes fixes it. looks intermittent

**Comment by mrxz** [comment]
Did a bit of digging, seems to be happening with Firefox 115. Locally I'm able to reproduce the test failures with Firefox 115, but not with 114. The following code in A-Frame is responsible for setting the `onload` and `onerror` methods (https://github.com/aframevr/aframe/blob/master/src/core/a-assets.js#L52-L60). This code isn't always hit with Firefox 115 as sometimes `imgEl.complete` is already true. Probably caused by other tests using the same image src.

**Comment by dmarcos** [comment]
Interesting. Seems Firefox 115 behavior is somehow different

**Comment by dmarcos** [comment]
Not sure if related but we recently changed the behavior https://github.com/aframevr/aframe/commit/1773661e4aa41a095ba8af6da57b15f4561ba2b5 to address https://github.com/aframevr/aframe/issues/5251 

Is there anything relevant in Firefox 115 change log or any known related bugs?

**Comment by mrxz** [comment]
It's really a combination of that change (which causes `onload` to be possibly undefined) and some change in Firefox 115. I did quickly skim through the changelog but didn't notice anything relevant.

In any case, I've opened a PR to fix the test in question: #5331 
As far as I can tell, there should be no need to manually call `onload` as it can just wait for the image to naturally load (as is done in other tests as well).
